### PR TITLE
[Woo POS] Tap background to dismiss modals

### PR DESCRIPTION
**Pull request by joshheald**
Part of: #13912

## Description

This PR alllows some POS Modals to be dismissed by tapping the background overlay.

It also adds a modifier, `posInteractiveDismissDisabled(disabled:)`, which works the same way as the built in `interactiveDismissDisabled` modifier does for sheets. This is used to prevent card reader connection modals, other than the `complete` modal, from being dismissed without calling the cancellation handler, or where cancellation isn't possible.

A future PR will allow tap to dismiss for all the dismissable connection modals, but more supporting work is required for this.

@staskus @iamgabrielma Only needs one review, and feel free to bump it to 20.5 if needed.

## Steps to reproduce

Test the following modals:

- [x] Exit POS confirmation
- [x] Simple Products information
- [x] Payment Capture error information (see #13642 for instructions to show this)
- [x] Card reader connection success modal

They should all be dismissable by tapping the background.

Test that all other card reader connection modals can only be dismissed by tapping their `x` button, where present.

## Testing information

I have tested the above scenarios on an iPad running iOS 16, and a simulator running iOS 17.

No unit tests as this is View-only code

## Screenshots

https://github.com/user-attachments/assets/5393d992-2d4b-449a-8e18-05ddfdc2fbb5

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [x] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [x] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [x] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on all devices (phone/tablet) and no regressions are added.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot or video is included in the pull request. Consider adding some for clarity.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/Classes/POS/Presentation/Card Present Payments/Connection Alerts/PointOfSaleCardPresentPaymentAlertType.swift [hunk 1]
```diff
@@ -25,4 +25,12 @@ enum PointOfSaleCardPresentPaymentAlertType: Hashable, Identifiable {
     case connectingFailedUpdateAddress(viewModel: PointOfSaleCardPresentPaymentConnectingFailedUpdateAddressAlertViewModel)
     case connectingFailedUpdatePostalCode(viewModel: PointOfSaleCardPresentPaymentConnectingFailedUpdatePostalCodeAlertViewModel)
     case connectionSuccess(viewModel: PointOfSaleCardPresentPaymentConnectionSuccessAlertViewModel)
+
+    var isNonDismissable: Bool {
```
Comment (iamgabrielma) [inline WooCommerce/Classes/POS/Presentation/Card Present Payments/Connection Alerts/PointOfSaleCardPresentPaymentAlertType.swift]: Just a nit, and this could be entirely an english-as-non-first-language thing so feel free to ignore: 

Is it more understandable by renaming it to `isDismissDisabled`? Or switching the modifier to `isDimissable(true/false)`? I find the combination of posInteractiveDismissDisabled + isNonDismissable a bit hard to follow at first.

File: WooCommerce/Classes/POS/Presentation/Reusable Views/POSModalManager.swift [hunk 2]
```diff
@@ -2,18 +2,33 @@ import SwiftUI
 
 class POSModalManager: ObservableObject {
     @Published private(set) var isPresented: Bool = false
+    @Published var allowsInteractiveDismissal: Bool = true
```
Comment (iamgabrielma) [inline WooCommerce/Classes/POS/Presentation/Reusable Views/POSModalManager.swift]: I think this can be `private(set)`:
```suggestion
    @Published private(set) var allowsInteractiveDismissal: Bool = true
```

File: WooCommerce/Classes/POS/ViewModels/TotalsViewModel.swift [hunk 3]
```diff
@@ -20,7 +20,6 @@ final class TotalsViewModel: ObservableObject, TotalsViewModelProtocol {
         case cardPaymentSuccessful
     }
 
-    @Published var showsCardReaderSheet: Bool = false
```
Comment (iamgabrielma) [inline WooCommerce/Classes/POS/ViewModels/TotalsViewModel.swift:23]: 💯

**Review by iamgabrielma — APPROVED** [review]
Tests well and work great, tested on 17.5 iPad air simulator, and 17.6.1 iPad Pro physical device  🚢

File: WooCommerce/Classes/POS/Presentation/Card Present Payments/Connection Alerts/PointOfSaleCardPresentPaymentAlertType.swift (hunk 1 above)
Comment (joshheald) [reply WooCommerce/Classes/POS/Presentation/Card Present Payments/Connection Alerts/PointOfSaleCardPresentPaymentAlertType.swift]: I want to keep the modifier matching the convention that Apple have with `isInteractiveDismissDisabled` for sheets.

`isDismissDisabled` is fine with me though – "is non-dismissable" reads more as a fixed property of the object, rather than something that might change. In this iteration, it's fairly fixed, but it it's going to change more in future.

File: WooCommerce/Classes/POS/Presentation/Reusable Views/POSModalManager.swift (hunk 2 above)
Comment (joshheald) [reply WooCommerce/Classes/POS/Presentation/Reusable Views/POSModalManager.swift]: 5cfb3d6
Good catch, I used it in a binding briefly and forgot to change it back when that didn't work.

File: WooCommerce/Classes/POS/Presentation/Card Present Payments/Connection Alerts/PointOfSaleCardPresentPaymentAlertType.swift (hunk 1 above)
Comment (joshheald) [reply WooCommerce/Classes/POS/Presentation/Card Present Payments/Connection Alerts/PointOfSaleCardPresentPaymentAlertType.swift]: 8024ffa

**Comment by iamgabrielma** [comment]
I'll go ahead and merge this!
